app/presenters/idv/by_mail/letter_requested_email_presenter.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Idv
+  module ByMail
+    class LetterRequestedEmailPresenter
+      include ActionView::Helpers::TranslationHelper
+      include Rails.application.routes.url_helpers
+
+      attr_reader :current_user, :service_provider
+      def initialize(current_user:)
+        @current_user = current_user
+        @service_provider = current_user.pending_profile&.initiating_service_provider
+      end
+
+      def sp_name
+        service_provider&.friendly_name
+      end
+
+      def show_sp_contact_instructions?
+        sp_name.present?
+      end
+
+      def show_sign_in_cta?
+        sign_in_url.present?
+      end
+
+      def sign_in_url
+        if service_provider.present?
+          service_provider_homepage_url
+        else
+          root_url
+        end
+      end
+
+      def url_options
+        {}
+      end
```

Comment:
I haven't tested, but part of the purpose of `url_options` here is to ensure that we maintain the intended locale with links. I'd be concerned if hard-coding an empty hash (vs. passing `url_options` with locale from mailer class) that the generated links may always be in English, even if a user's preferred email communications language is different.

Reply:
Good catch, we pulled that pattern from [AccountVerifiedEmailPresenter](https://github.com/18F/identity-idp/blob/ca3ed1ac06e06cd7c9dcafa82bc653c8652da2f0/app/presenters/idv/account_verified_email_presenter.rb#L33-L35). We probably want to clean it up in both places.

It looks like `VerificationResultsEmailPresenter` has an example for how to pass the url options from `UserMailer`.

Reply:
Thanks for catching that, Andrew! Will fix that and update it.

Reply:
Updated! (Also fixed the AccountVerifiedEmailPresenter)